src/librustc/traits/select.rs

```diff
@@ -2084,6 +2083,22 @@ impl<'cx, 'gcx, 'tcx> SelectionContext<'cx, 'gcx, 'tcx> {
                 Where(ty::Binder(tys.to_vec()))
             }
 
+            ty::TyClosure(def_id, substs) => {
+                let trait_id = obligation.predicate.def_id();
+                let copy_closures =
```

Comment:
Don't you need to check the feature-gate status of the crate that defined the closure, to avoid ICEs?

Crate A:
```Rust
#![feature(clone_closures)]

pub trait Generic {
    fn foo<T: Clone>(self, t: T);
}

pub fn example<G: Generic>(g: G) {
    g.foo(|| {});
}
```

Crate B:
```Rust
// no feature flags!

extern crate a;

struct S;
impl Generic for S {
    fn foo<T: Clone>(self, t: T) { let _ = t.clone(); }
}

fn main() {
    // trans will monomorphize this, and will try to evaluate `[closure]: Clone`, which
    // will fail and cause an ICE.
    foo::example(S);
}
```

Another option would be to make all closures `Copy` & `Clone` in trans, but that would be visible through specialization etc. so I'm not sure it's the better idea.

Reply:
@arielb1 Yes you're right. I'm working on it.